examples/keyberon48.rs

```diff
@@ -0,0 +1,230 @@
+use stm32f1xx_hal::{gpio, pac, timer};
+use usb_device::bus::UsbBusAllocator;
+use usb_device::class::UsbClass as _;
+
+type UsbClass = keyberon::Class<'static, UsbBusType, Leds>;
+type UsbDevice = keyberon::Device<'static, UsbBusType>;
+
+pub struct Leds {
+	caps_lock: gpio::gpioc::PC13<gpio::Output<gpio::PushPull>>,
+}
+impl keyberon::keyboard::Leds for Leds {
+	fn caps_lock(&mut self, status: bool) {
+		if status {
+			self.caps_lock.set_low().unwrap()
+		} else {
+			self.caps_lock.set_high().unwrap()
+		}
+	}
+}
+
+pub struct Cols(
+	pub PB12<Input<PullUp>>,
+	pub PB13<Input<PullUp>>,
+	pub PB14<Input<PullUp>>,
+	pub PB15<Input<PullUp>>,
+	pub PA8<Input<PullUp>>,
+	pub PA9<Input<PullUp>>,
+	pub PA10<Input<PullUp>>,
+	pub PB5<Input<PullUp>>,
+	pub PB6<Input<PullUp>>,
+	pub PB7<Input<PullUp>>,
+	pub PB8<Input<PullUp>>,
+	pub PB9<Input<PullUp>>,
+);
+impl_heterogenous_array! {
+	Cols,
+	dyn InputPin<Error = Infallible>,
+	U12,
+	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+}
+
+pub struct Rows(
+	pub PB11<Output<PushPull>>,
+	pub PB10<Output<PushPull>>,
+	pub PB1<Output<PushPull>>,
+	pub PB0<Output<PushPull>>,
+);
+impl_heterogenous_array! {
+	Rows,
+	dyn OutputPin<Error = Infallible>,
+	U4,
+	[0, 1, 2, 3]
+}
+
+const CUT: Action = m(&[LShift, Delete]);
+const COPY: Action = m(&[LCtrl, Insert]);
+const PASTE: Action = m(&[LShift, Insert]);
+const C_ENTER: Action = HoldTap {
+	timeout: 200,
+	hold: &k(LCtrl),
+	tap: &k(Enter),
+};
+const L1_SP: Action = HoldTap {
+	timeout: 200,
+	hold: &l(1),
+	tap: &k(Space),
+};
+
+#[rustfmt::skip]
+pub static LAYERS: keyberon::layout::Layers = &[
+	&[
+		&[k(Escape)	,k(Q)			,k(W)	,k(E)		,k(R)	,k(T)	,k(Y),     k(U)	,k(I)	,k(O)		,k(P)		,k(LBracket)],
+		&[k(Tab)	,k(A)			,k(S)	,k(D)		,k(F)	,k(G)	,k(H),     k(J)	,k(K)	,k(L)		,k(SColon)	,k(Quote)   ],
+		&[k(LShift)	,k(NonUsBslash)	,k(Z)	,k(X)		,k(C)	,k(V)	,k(B),     k(N)	,k(M)	,k(Comma)	,k(Dot)		,k(Slash)	],
+		&[Trans		,Trans			,k(LAlt),k(LGui)	,C_ENTER,l(2)	,k(BSpace),L1_SP,k(RAlt),Trans		,Trans		,Trans		],
+	], &[
+		&[k(Grave)	,Trans	,Trans		,Trans	,Trans		,k(PgUp)			,k(PgDown)		,k(Home),k(Up)	,k(End)		,k(Minus)	,k(Equal)	],
+		&[Trans		,Trans	,Trans		,Trans	,Trans		,k(MediaVolDown)	,k(MediaVolUp)	,k(Left),k(Down),k(Right)	,Trans		,k(RBracket)],
+		&[Trans		,k(Kb1)	,k(Kb2)		,k(Kb3)	,k(Kb4)		,k(Kb5)				,k(Kb6)			,k(Kb7)	,k(Kb8)	,k(Kb9)		,k(Kb0)		,k(Bslash)	],
+		&[Trans		,Trans	,Trans		,Trans	,Trans		,Trans				,Trans			,Trans	,Trans	,Trans		,Trans		,Trans		],
+	], &[
+		&[Trans		,k(F1)	,k(F2)	,k(F3)	,k(F4)	,Trans	,Trans		,m(&[RAlt, Kb8])	,m(&[RAlt, Kb9])	,Trans				,Trans	,Trans ],
+		&[Trans		,k(F5)	,k(F6)	,k(F7)	,k(F8)	,Trans	,Trans		,m(&[RAlt, Kb7])	,m(&[RAlt, Kb0])	,Trans				,Trans	,Trans ],
+		&[Trans		,k(F9)	,k(F10)	,k(F11)	,k(F12)	,Trans	,Trans		,Trans				,m(&[LShift, Kb8])	,m(&[LShift, Kb9])	,Trans	,Trans ],
+		&[Trans		,Trans	,Trans	,Trans	,Trans	,Trans	,Trans		,Trans				,Trans				,Trans				,Trans	,Trans ],
+	],
+];
```

Comment:
Why not have the function keys overlapping with the digits but on a different layer? E.g.  k(1) on one layer and K(F1) one layer above it?

Reply:
These kind of things are really personal preference. I would not like your idea personally because you then can't put all the function keys on the line.

Reply:
> These kind of things are really personal preference. I would not like your idea personally because you then can't put all the function keys on the line.

@gilescope that was my first version but I struggled with that, so it ended up in this version.